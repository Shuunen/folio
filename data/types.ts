export interface Photo {
  label: string
  size: string
  src: string
  thumb: string
}
