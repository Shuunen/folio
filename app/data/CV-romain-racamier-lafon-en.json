{
  "basics": {
    "email": "romain.racamier@gmail.com",
    "label": "Web Developer - Tech Lead",
    "name": "Romain Racamier-Lafon"
  },
  "education": [
    {
      "endDate": "2020-01-31",
      "institution": "Unow",
      "startDate": "2020-07-01",
      "studyType": "Business tutor"
    },
    {
      "endDate": "2011-09-01",
      "institution": "IIA Laval",
      "startDate": "2009-09-01",
      "studyType": "Master in Computer Engineering"
    },
    {
      "endDate": "2009-09-01",
      "institution": "IUT Laval",
      "startDate": "2007-09-01",
      "studyType": "DUT Communication Services and Networks"
    },
    {
      "endDate": "2007-09-01",
      "institution": "Montesquieu High School",
      "startDate": "2005-09-01",
      "studyType": "Scientific Baccalaureate"
    }
  ],
  "interests": [
    {
      "name": "Volunteering, support"
    },
    {
      "name": "Photo, digital editing"
    },
    {
      "name": "Design/logo creation"
    },
    {
      "name": "Hiking"
    }
  ],
  "skills": [
    {
      "keywords": [
        "Typescript, Javascript, ES6+",
        "React, Next, Material-UI",
        "Vue 2, Vue 3, Nuxt, Vuetify",
        "Angular, AngularJs",
        "TailwindCSS, Bootstrap, CSS3",
        "HTML 5, HTML 4, XHTML",
        "Web Components, PWA",
        "Jest, Vitest, Mocha, C8",
        "NodeJs, ExpressJs",
        "PHP, MySQL, SQLite"
      ],
      "name": "Development"
    },
    {
      "keywords": [
        "Google Lighthouse, YSlow",
        "Eslint, Biome, Prettier",
        "Npm, Yarn, Pnpm",
        "Vite, EsBuild, Webpack",
        "Git, GitFlow, GitHub, GitLab"
      ],
      "name": "Tools"
    },
    {
      "keywords": [
        "Français : mother tongue",
        "English: confirmed & technical (professional and personal trips abroad)"
      ],
      "name": "Languages"
    },
    {
      "keywords": [
        "Scrum, Agile",
        "Active listening, NLP",
        "TDD, BDD, Gherkin"
      ],
      "name": "Methodologies"
    }
  ],
  "work": [
    {
      "company": "Collectif Energie",
      "endDate": "2024-09-01",
      "highlights": [
        "Tech Lead in a team of 6 people",
        "Developed new features and maintained a Nuxt 3 application",
        "Created a monorepo to streamline development, centralize libraries, and align practices",
        "Migrated existing sites and libraries to the monorepo",
        "Built a React component library based on ShadCN, RadixUI & Tailwind CSS",
        "Implemented and participated in the recruitment process",
        "Environment : TypeScript, Nx, Vue 3, Nuxt 3, React 18, NestJs, Vite, Vitest, Tailwind, NodeJs 20, EsLint, Git"
      ],
      "position": "Web Developer - Tech Lead",
      "startDate": "2024-01-01"
    },
    {
      "company": "WeSave",
      "endDate": "2023-12-01",
      "highlights": [
        "Co-Tech Lead in a team of 4 people",
        "Migrated a Nuxt project from JS & Vue 2 to TypeScript & Vue 2.7",
        "Developed new features and maintained the main application",
        "Created a custom ESLint plugin",
        "Participated in the recruitment process",
        "Environment : TypeScript, Vue 2, Vue 2.7, Nuxt 2, HTML 5, CSS 3, Tailwind, NodeJS 18, EsLint, Git"
      ],
      "position": "Web Developer - Tech Lead",
      "startDate": "2022-09-01"
    },
    {
      "company": "Euro Information (en mission pour GFI / Inetum)",
      "endDate": "2022-09-01",
      "highlights": [
        "Built a component library to centralize the UI/UX of all current and future web projects",
        "Redesigned the Lyf.eu website from AngularJs to Nuxt using Vue & Tailwind CSS",
        "Developed new features and maintained the internal framework used to manage Lyf Pay application data",
        "Mentored and trained my intern on an activity tracking project",
        "Environment : Angular 6 à 13, Vue 2, Nuxt 2, TypeScript, HTML 5, CSS 3, Tailwind, NodeJS 8 to 14, Git"
      ],
      "position": "Web Developer - Tech Lead & Mentor",
      "startDate": "2018-09-01"
    },
    {
      "company": "Niji",
      "endDate": "2018-06-01",
      "highlights": [
        "For Auchan: Created a chatbot with DialogFlow & VueJS for online drive visitors",
        "For Cipav: Redesigned the extranet in Angular for members",
        "For Enedis: Maintained an Ionic application for iOS & Android",
        "For Verlingue Immobilier: Created a professional subscription portal",
        "For Système U: Created  a management web-app for employees",
        "Environment : TypeScript, Angular 4, Vue 2,  HTML 5, CSS 3, Git"
      ],
      "position": "Web Developer - Lead Dev",
      "startDate": "2016-10-01"
    },
    {
      "company": "Amdocs (en mission pour Niji)",
      "endDate": "2016-09-01",
      "highlights": [
        "Creation of various internal web applications aimed at managing a catalog of telecoms offers",
        "Front-end lead on the main project which lasted 2 years",
        "Creation of a building interface using blocks and drag & drop",
        "Setting up unit tests and E2E tests via PhantomJS",
        "Environment : JavaScript, BackboneJS, JQuery, RequireJS, HTML5, CSS3, Sass, Perforce"
      ],
      "position": "Web Developer - Tech Lead",
      "startDate": "2014-04-01"
    },
    {
      "company": "Ouest-France (en mission pour Alyotech)",
      "endDate": "2014-03-01",
      "highlights": [
        "Front-end lead during the redesign of the Ouest-France website",
        "Creation of 3 Drupal themes: front, back and mobile",
        "Performance and SEO optimization",
        "Environment : JavaScript, JQuery, JQuery UI, Bootstrap 2, Drupal 7, HTML5, CSS3, Google Analytics, Google Tag Manager, Git"
      ],
      "position": "Web Developer - Tech Lead",
      "startDate": "2012-11-01"
    },
    {
      "company": "Alyotech",
      "endDate": "2012-10-01",
      "highlights": [
        "Redesign of the SINOE website for ADEME",
        "Creation of front pages and modules",
        "Maintaining the AccessiWeb Bronze specifications (like WAI-ARIA)",
        "Environment : Zend PHP, Javascript, JQuery, XHTML 1.0, CSS 2, SVN"
      ],
      "position": "Web Developer",
      "startDate": "2011-10-01"
    },
    {
      "company": "Jouve",
      "endDate": "2011-09-01",
      "highlights": [
        "Creation of page and modules for ANTS",
        "Developments for the Defense clubs federation",
        "Environment : Cake PHP, Symfony, Javascript, HTML, CSS 2, SVN"
      ],
      "position": "Web Developer",
      "startDate": "2009-09-01"
    },
    {
      "company": "Sephira",
      "endDate": "2008-05-01",
      "highlights": [
        "Fixes on the embedded software of the Intellio vital card reader",
        "Deployment and integration of a data processing service",
        "Environment : C, C++, Bash, Linux, SVN"
      ],
      "position": "Software Developer",
      "startDate": "2008-03-01"
    }
  ]
}
