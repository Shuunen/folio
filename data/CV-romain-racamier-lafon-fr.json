{
  "basics": {
    "email": "romain.racamier@gmail.com",
    "label": "Développeur Web - Lead Tech",
    "name": "Romain Racamier-Lafon",
    "summary": "Curieux et motivé j'aime l’innovation, la création et la dynamique de groupe."
  },
  "education": [
    {
      "endDate": "2020-01-31",
      "institution": "Unow",
      "startDate": "2020-07-01",
      "studyType": "Tuteur en entreprise"
    },
    {
      "endDate": "2011-09-01",
      "institution": "IIA Laval",
      "startDate": "2009-09-01",
      "studyType": "Master en Ingénierie Informatique"
    },
    {
      "endDate": "2009-09-01",
      "institution": "IUT Laval",
      "startDate": "2007-09-01",
      "studyType": "DUT Service et Réseaux de Communication"
    },
    {
      "endDate": "2007-09-01",
      "institution": "Lycée Montesquieu",
      "startDate": "2005-09-01",
      "studyType": "Bac Scientifique"
    }
  ],
  "interests": [
    {
      "name": "Bénévolat, accompagnement"
    },
    {
      "name": "Photo, macro et retouche"
    },
    {
      "name": "Création de design/logos"
    },
    {
      "name": "Randonnée"
    }
  ],
  "skills": [
    {
      "keywords": [
        "Typescript, Javascript, ES6+",
        "React, Next, Material-UI",
        "Vue 2, Vue 3, Nuxt, Vuetify",
        "Angular, AngularJs",
        "TailwindCSS, Bootstrap, CSS3",
        "HTML 5, HTML 4, XHTML",
        "Web Component, PWA",
        "Jest, Vitest, Mocha, C8",
        "NodeJs, ExpressJs",
        "PHP, MySQL, SQLite"
      ],
      "name": "Développement"
    },
    {
      "keywords": [
        "Google Lighthouse, YSlow",
        "Eslint, Biome, Prettier",
        "Npm, Yarn, Pnpm",
        "Vite, EsBuild, Webpack",
        "Git, GitFlow, GitHub, GitLab"
      ],
      "name": "Outillage"
    },
    {
      "keywords": [
        "Français : langue maternelle",
        "Anglais : confirmé & technique (voyages pro & perso à l'étranger)"
      ],
      "name": "Langues"
    },
    {
      "keywords": [
        "Scrum, Agile",
        "Écoute active, PNL",
        "TDD, BDD, Gherkin"
      ],
      "name": "Méthodologies"
    }
  ],
  "work": [
    {
      "company": "Collectif Energie",
      "endDate": "2024-09-01",
      "highlights": [
        "Lead tech dans une équipe de 6 personnes",
        "Ajout de fonctionnalités et maintenance d'une application Nuxt 3",
        "Création d'un monorepo pour industrialiser les développements, centraliser les librairies et aligner les pratiques",
        "Migrations des sites et libraries existantes au sein du monorepo",
        "Création d'un catalogue de composants React basés sur ShadCn, RadixUI & Tailwind CSS",
        "Mise en place et participation au processus de recrutement",
        "Environnement : TypeScript, Nx, Vue 3, Nuxt 3, React 18, NestJs, Vite, Vitest, Tailwind, NodeJs 20, EsLint, Git"
      ],
      "position": "Développeur Web - Lead Tech",
      "startDate": "2024-01-01"
    },
    {
      "company": "WeSave",
      "endDate": "2023-12-01",
      "highlights": [
        "Co-Lead tech dans une équipe de 4 personnes",
        "Migration du projet Nuxt : JS & Vue 2 vers TypeScript & Vue 2.7",
        "Ajout de fonctionnalités et maintenance de l'application principale",
        "Création d'un plugin EsLint custom",
        "Participation au processus de recrutement",
        "Environnement : TypeScript, Vue 2, Vue 2.7, Nuxt 2, HTML 5, CSS 3, Tailwind, NodeJS 18, EsLint, Git"
      ],
      "position": "Développeur Web - Lead Tech",
      "startDate": "2022-09-01"
    },
    {
      "company": "Euro Information (en mission pour GFI / Inetum)",
      "endDate": "2022-09-01",
      "highlights": [
        "Création d'un catalogue de composants visant à centraliser l'UI et l'UX de tous les projets web existants et futurs",
        "Refonte du site institutionnel Lyf.eu Angular en Nuxt avec Vue & Tailwind CSS",
        "Ajout de fonctionnalités et maintenance sur le framework interne de gestion des données liées à l'application Lyf Pay",
        "Accompagnement et formation de mon stagiaire autour d'un projet applicatif de suivi d'activité",
        "Environnement : Angular 6 à 13, Vue 2, Nuxt 2, TypeScript, HTML 5, CSS 3, Tailwind, NodeJS 8 à 14, Git"
      ],
      "position": "Développeur Web - Lead Tech & Tuteur",
      "startDate": "2018-09-01"
    },
    {
      "company": "Niji",
      "endDate": "2018-06-01",
      "highlights": [
        "Pour Auchan : Création d'un chatbot avec DialogFlow & VueJS pour les visiteurs du drive en ligne",
        "Pour la Cipav : Refonte de l'extranet en Angular pour les adhérents",
        "Pour Enedis : maintenance d'une application Ionic pour iOS & Android",
        "Pour Verlingue Immobilier : Création d'un espace de souscription pro",
        "Pour Système U :  Création d'une web-app de gestion pour les employés du drive",
        "Environnement : TypeScript, Angular 4, Vue 2,  HTML 5, CSS 3, Git"
      ],
      "position": "Développeur Web - Lead Tech",
      "startDate": "2016-10-01"
    },
    {
      "company": "Amdocs (en mission pour Niji)",
      "endDate": "2016-09-01",
      "highlights": [
        "Réalisation de différentes applications web internes visant à gérer un catalogue d'offres télécoms",
        "Responsable Front-End sur le projet principal qui a duré 2 ans",
        "Création d'une interface de construction par blocs et drag & drop",
        "Mise en place des tests unitaires et tests E2E via PhantomJS",
        "Environnement : JavaScript, BackboneJS, JQuery, RequireJS, HTML5, CSS3, Sass, Perforce"
      ],
      "position": "Développeur Web - Lead Tech",
      "startDate": "2014-04-01"
    },
    {
      "company": "Ouest-France (en mission pour Alyotech)",
      "endDate": "2014-03-01",
      "highlights": [
        "Responsable Front-End lors de la refonte du site Ouest-France",
        "Création de 3 thèmes Drupal : front, back et mobile",
        "Optimisation des performances et du référencement (SEO)",
        "Environnement : JavaScript, JQuery, JQuery UI, Bootstrap 2, Drupal 7, HTML5, CSS3, Google Analytics, Google Tag Manager, Git"
      ],
      "position": "Développeur Web - Lead Tech",
      "startDate": "2012-11-01"
    },
    {
      "company": "Alyotech",
      "endDate": "2012-10-01",
      "highlights": [
        "Refonte d'un site d'information pour l'ADEME : bit.ly/sinoe-web",
        "Création de pages et modules front",
        "Maintien du cahier des charges AccessiWeb Bronze : bit.ly/acc-web",
        "Environnement : Zend PHP, Javascript, JQuery, XHTML 1.0, CSS 2, SVN"
      ],
      "position": "Développeur Web",
      "startDate": "2011-10-01"
    },
    {
      "company": "Jouve",
      "endDate": "2011-09-01",
      "highlights": [
        "Création de page et modules pour l'ANTS",
        "Prise en charge d'évolutions pour la fédération des clubs de la Défense",
        "Environnement : Cake PHP, Symfony, Javascript, HTML, CSS 2, SVN"
      ],
      "position": "Développeur Web",
      "startDate": "2009-09-01"
    },
    {
      "company": "Sephira",
      "endDate": "2008-05-01",
      "highlights": [
        "Corrections sur le logiciel embarqué du lecteur de carte vitale Intellio",
        "Déploiement et intégration d'un service de traitement de données",
        "Environnement : C, C++, Bash, Linux, SVN"
      ],
      "position": "Développeur logiciel",
      "startDate": "2008-03-01"
    }
  ]
}
