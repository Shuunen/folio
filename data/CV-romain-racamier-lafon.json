{
  "$schema": "./cv.schema.json",
  "basics": {
    "email": "romain.racamier@gmail.com",
    "label": "Développeur Web - Lead Tech",
    "name": "Romain Racamier-Lafon",
    "summary": "Curieux et motivé j'aime l’innovation, la création et la dynamique de groupe."
  },
  "education": [
    {
      "endDate": "2020-01-31",
      "institution": "Unow",
      "startDate": "2020-07-01",
      "studyType": "Tuteur en entreprise"
    },
    {
      "endDate": "2011-09-01",
      "institution": "IIA Laval",
      "startDate": "2009-09-01",
      "studyType": "Master en Ingénierie Informatique"
    },
    {
      "endDate": "2009-09-01",
      "institution": "IUT Laval",
      "startDate": "2007-09-01",
      "studyType": "DUT Service et Réseaux de Communication"
    },
    {
      "endDate": "2007-09-01",
      "institution": "Lycée Montesquieu",
      "startDate": "2005-09-01",
      "studyType": "Bac Scientifique"
    }
  ],
  "interests": [
    {
      "name": "Bénévolat, accompagnement"
    },
    {
      "name": "Photo, macro et retouche"
    },
    {
      "name": "Création de design/logos"
    },
    {
      "name": "Randonnée"
    }
  ],
  "skills": [
    {
      "keywords": [
        "Typescript, Javascript, ES6+",
        "React, Next, Material-UI",
        "Vue 2, Vue 3, Nuxt, Vuetify",
        "Angular, AngularJs",
        "TailwindCSS, Bootstrap, CSS3",
        "HTML 5, HTML 4, XHTML",
        "Web Component, PWA",
        "Jest, Vitest, Mocha, C8",
        "NodeJs, ExpressJs",
        "PHP, MySQL, SQLite"
      ],
      "name": "Développement"
    },
    {
      "keywords": [
        "Google Lighthouse, YSlow",
        "Eslint, Biome, Prettier",
        "Npm, Yarn, Pnpm",
        "Vite, EsBuild, Webpack",
        "Git, GitFlow, GitHub, GitLab"
      ],
      "name": "Outillage"
    },
    {
      "keywords": [
        "Français : langue maternelle",
        "Anglais : confirmé & technique (voyages pro & perso à l'étranger)"
      ],
      "name": "Langues"
    },
    {
      "keywords": [
        "Scrum, Agile",
        "Écoute active, PNL",
        "TDD, BDD, Gherkin"
      ],
      "name": "Méthodologies"
    }
  ],
  "work": [
    {
      "company": "BatVoice AI",
      "endDate": "2024-02-15",
      "highlights": [],
      "keywords": [
        "TypeScript",
        "Vue 2",
        "Vue 3",
        "Vite",
        "Vitest",
        "Tailwind",
        "NodeJS"
      ],
      "photos": [
        {
          "label": "BatVoice AI",
          "size": "1459-1018",
          "src": "public:/images/batvoice-ai-01.svg",
          "thumb": "public:/images/batvoice-ai-01.svg"
        }
      ],
      "position": "dev-web-tl",
      "sectors": [
        "voice",
        "ai"
      ],
      "startDate": "2024-11-15"
    },
    {
      "company": "Collectif Energie",
      "endDate": "2024-09-01",
      "highlights": [
        "Lead tech dans une équipe de 6 personnes",
        "Ajout de fonctionnalités et maintenance d'une application Nuxt 3",
        "Création d'un monorepo pour industrialiser les développements, centraliser les librairies et aligner les pratiques",
        "Migrations des sites et libraries existantes au sein du monorepo",
        "Création d'un catalogue de composants React basés sur ShadCn, RadixUI & Tailwind CSS",
        "Mise en place et participation au processus de recrutement"
      ],
      "keywords": [
        "TypeScript",
        "Nx",
        "Vue 3",
        "Nuxt 3",
        "React",
        "NestJs",
        "Vite",
        "Vitest",
        "Tailwind",
        "NodeJS"
      ],
      "photos": [
        {
          "label": "Collectif Energie",
          "size": "1440-1018",
          "src": "public:/images/collectif-energie-01.svg",
          "thumb": "public:/images/collectif-energie-01.svg"
        }
      ],
      "position": "dev-web-tl",
      "sectors": [
        "energy",
        "savings-management"
      ],
      "startDate": "2024-01-01"
    },
    {
      "company": "WeSave",
      "endDate": "2023-12-01",
      "highlights": [
        "Co-Lead tech dans une équipe de 4 personnes",
        "Migration du projet Nuxt : JS & Vue 2 vers TypeScript & Vue 2.7",
        "Ajout de fonctionnalités et maintenance de l'application principale",
        "Création d'un plugin EsLint custom",
        "Participation au processus de recrutement"
      ],
      "keywords": [
        "TypeScript",
        "Nuxt",
        "Vue 3",
        "Vue 2",
        "HTML",
        "CSS",
        "Tailwind",
        "NodeJS"
      ],
      "photos": [
        {
          "label": "WeSave conseil",
          "size": "1440-998",
          "src": "public:/images/wesave-conseil-01.webp",
          "thumb": "public:/images/wesave-conseil-01-w410.webp"
        }
      ],
      "position": "dev-web-tl",
      "sectors": [
        "finance",
        "savings-management"
      ],
      "startDate": "2022-07-01"
    },
    {
      "company": "Lyf, Euro Information, GFI, Inetum",
      "endDate": "2022-09-01",
      "highlights": [
        "Création d'un catalogue de composants visant à centraliser l'UI et l'UX de tous les projets web existants et futurs",
        "Refonte du site institutionnel Lyf.eu Angular en Nuxt avec Vue & Tailwind CSS",
        "Ajout de fonctionnalités et maintenance sur le framework interne de gestion des données liées à l'application Lyf Pay",
        "Accompagnement et formation de mon stagiaire autour d'un projet applicatif de suivi d'activité",
        "Environnement : Angular 6 à 13, Vue 2, Nuxt 2, TypeScript, HTML 5, CSS 3, Tailwind, NodeJS 8 à 14, Git"
      ],
      "keywords": [
        "TypeScript",
        "Nuxt",
        "Vue 2",
        "Angular",
        "HTML",
        "CSS",
        "Tailwind",
        "NodeJS",
        "Git"
      ],
      "photos": [
        {
          "label": "Lyf.eu - Page accueil",
          "size": "1701-1079",
          "src": "public:/images/lyf-01.webp",
          "thumb": "public:/images/lyf-01-w410.webp"
        },
        {
          "label": "Lyf.eu - Page smartphone",
          "size": "1720-1657",
          "src": "public:/images/lyf-02.webp",
          "thumb": "public:/images/lyf-02-w410.webp"
        }
      ],
      "position": "dev-web-tl-tutor",
      "sectors": [
        "finance",
        "mobile-payment"
      ],
      "startDate": "2018-09-01"
    },
    {
      "company": "Auchan, Enedis, Système U, Niji",
      "endDate": "2018-06-01",
      "highlights": [
        "Pour Auchan : Création d'un chatbot avec DialogFlow & VueJS pour les visiteurs du drive en ligne",
        "Pour la Cipav : Refonte de l'extranet en Angular pour les adhérents",
        "Pour Enedis : maintenance d'une application Ionic pour iOS & Android",
        "Pour Verlingue Immobilier : Création d'un espace de souscription pro",
        "Pour Système U :  Création d'une web-app de gestion pour les employés du drive",
        "Environnement : TypeScript, Angular 4, Vue 2,  HTML 5, CSS 3, Git"
      ],
      "keywords": [
        "TypeScript",
        "Angular",
        "Vue 2",
        "AngularJs",
        "HTML",
        "CSS",
        "Sass",
        "Git"
      ],
      "photos": [
        {
          "label": "Enedis - Présentation smartphone",
          "size": "4096-2730",
          "src": "public:/images/enedis-01.webp",
          "thumb": "public:/images/enedis-01-w410.webp"
        }
      ],
      "position": "dev-web-tl",
      "sectors": [
        "retail",
        "energy",
        "insurance"
      ],
      "startDate": "2016-10-01"
    },
    {
      "company": "Optus, Streamezzo, Amdocs, Niji",
      "endDate": "2016-09-01",
      "highlights": [
        "Réalisation de différentes applications web internes visant à gérer un catalogue d'offres télécoms",
        "Responsable Front-End sur le projet principal qui a duré 2 ans",
        "Création d'une interface de construction par blocs et drag & drop",
        "Mise en place des tests unitaires et tests E2E via PhantomJS",
        "Environnement : JavaScript, BackboneJS, JQuery, RequireJS, HTML5, CSS3, Sass, Perforce"
      ],
      "keywords": [
        "JavaScript",
        "BackboneJS",
        "JQuery",
        "RequireJS",
        "HTML",
        "CSS",
        "Sass",
        "Perforce"
      ],
      "photos": [
        {
          "label": "Optus - Interface smartphone",
          "size": "1369-772",
          "src": "public:/images/optus-01.webp",
          "thumb": "public:/images/optus-01-w410.webp"
        }
      ],
      "position": "dev-web-tl",
      "sectors": [
        "telecoms",
        "mobile"
      ],
      "startDate": "2014-04-01"
    },
    {
      "company": "Ouest-France, Alyotech",
      "endDate": "2014-03-01",
      "highlights": [
        "Responsable Front-End lors de la refonte du site Ouest-France",
        "Création de 3 thèmes Drupal : front, back et mobile",
        "Optimisation des performances et du référencement (SEO)",
        "Environnement : JavaScript, JQuery, JQuery UI, Bootstrap 2, Drupal 7, HTML5, CSS3, Google Analytics, Google Tag Manager, Git"
      ],
      "keywords": [
        "JavaScript",
        "Bootstrap 2",
        "HTML",
        "CSS",
        "Google Analytics",
        "Google Tag Manager",
        "Git"
      ],
      "photos": [
        {
          "label": "Ouest-France - Multi-écrans",
          "size": "1551-790",
          "src": "public:/images/ouest-france-01.webp",
          "thumb": "public:/images/ouest-france-01-w410.webp"
        }
      ],
      "position": "dev-web-tl",
      "sectors": [
        "press",
        "media"
      ],
      "startDate": "2012-11-01"
    },
    {
      "company": "ADEME, Alyotech",
      "endDate": "2012-10-01",
      "highlights": [
        "Refonte d'un site d'information pour l'ADEME : bit.ly/sinoe-web",
        "Création de pages et modules front",
        "Maintien du cahier des charges AccessiWeb Bronze : bit.ly/acc-web",
        "Environnement : Zend PHP, Javascript, JQuery, XHTML 1.0, CSS 2, SVN"
      ],
      "keywords": [
        "Zend PHP",
        "JavaScript",
        "JQuery",
        "HTML",
        "CSS",
        "SVN"
      ],
      "photos": [
        {
          "label": "Agence De l'Environnement et de la Maîtrise de l'Énergie - Page d'accueil",
          "size": "1260-800",
          "src": "public:/images/ademe-01.webp",
          "thumb": "public:/images/ademe-01-w410.webp"
        }
      ],
      "position": "dev-web",
      "sectors": [
        "environment",
        "energy"
      ],
      "startDate": "2011-10-01"
    },
    {
      "company": "ANTS, FCSAD, Jouve",
      "endDate": "2011-09-01",
      "highlights": [
        "Création de page et modules pour l'ANTS",
        "Prise en charge d'évolutions pour la fédération des clubs de la Défense",
        "Environnement : Cake PHP, Symfony, Javascript, HTML, CSS 2, SVN"
      ],
      "keywords": [
        "Cake PHP",
        "Symfony",
        "JavaScript",
        "HTML",
        "CSS",
        "SVN"
      ],
      "photos": [
        {
          "label": "Agence Nationale des Titres Sécurisés - Page d'accueil",
          "size": "1121-629",
          "src": "public:/images/ants-01.webp",
          "thumb": "public:/images/ants-01-w410.webp"
        },
        {
          "label": "Agence Nationale des Titres Sécurisés - Page d'accueil",
          "size": "1121-1638",
          "src": "public:/images/ants-02.webp",
          "thumb": "public:/images/ants-01-w410.webp"
        }
      ],
      "position": "dev-web",
      "sectors": [
        "government",
        "sport",
        "matriculation"
      ],
      "startDate": "2009-09-01"
    },
    {
      "company": "Sephira",
      "endDate": "2008-05-01",
      "highlights": [
        "Corrections sur le logiciel embarqué du lecteur de carte vitale Intellio",
        "Déploiement et intégration d'un service de traitement de données",
        "Environnement : C, C++, Bash, Linux, SVN"
      ],
      "keywords": [
        "C",
        "C++",
        "Bash",
        "Linux",
        "SVN"
      ],
      "photos": [
        {
          "label": "Sephira - Terminal Intellio",
          "size": "1184-1638",
          "src": "public:/images/sephira-intellio-01.webp",
          "thumb": "public:/images/sephira-intellio-01-w410.webp"
        }
      ],
      "position": "software-dev",
      "sectors": [
        "health",
        "insurance"
      ],
      "startDate": "2008-03-01"
    }
  ]
}
