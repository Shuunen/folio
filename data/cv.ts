import { clone, slugify } from 'shuutils'
import type { DeepReadonly } from 'vue'
import rawCv from './CV-romain-racamier-lafon.json'
import type { Photo } from './types'

type JsonResume = DeepReadonly<typeof rawCv>

type ResumeEducation = DeepReadonly<{ id: string } & JsonResume['education'][0]>

type ResumeWork = DeepReadonly<{
  company: string
  endDate: string
  highlights: string[]
  id: string
  keywords: string[]
  photos: Photo[]
  position: string
  sectors: string[]
  startDate: string
}>

type Resume = DeepReadonly<{
  education: ResumeEducation[]
  work: ResumeWork[]
}>

/**
 * Clean a string
 * @param input - String to clean
 * @returns Cleaned string
 */
function cleanString (input: string) {
  return input
    .trim()
    .replace(/<[^>]*>?/gu, '')
    .replace(/\s+/gu, ' ')
}

/**
 * Clean an array
 * @param input - Array to clean
 * @returns Cleaned array
 */
function cleanArray (input: Readonly<Readonly<Record<string, unknown>>[]> | Readonly<Readonly<string>[]>) {
  return input.map((item) => {
    if (typeof item === 'string') return cleanString(item)
    if (typeof item === 'object') return cleanObject(item) // eslint-disable-line @typescript-eslint/no-use-before-define
    return item
  })
}

/**
 * Clean an object
 * @param input - Object to clean
 * @returns Cleaned object
 */
export function cleanObject (input: Readonly<Record<string, unknown>>) {
  const data = clone(input)
  for (const [key, item] of Object.entries(data)) {
    if (typeof item === 'string') data[key] = cleanString(item)
    if (typeof item === 'object' && item) data[key] = cleanObject(item as Record<string, unknown>) // eslint-disable-line @typescript-eslint/consistent-type-assertions
    if (Array.isArray(item)) data[key] = cleanArray(item)
  }
  return data
}

/**
 * Set IDs
 * @param input - Input data
 * @returns Data with IDs
 */
export function setIds (input: JsonResume) {
  const data: Resume = {
    education: input.education.map((item) => ({
      ...item,
      id: slugify(item.institution),
    })),
    work: input.work.map((item) => ({
      company: item.company,
      endDate: item.endDate ?? '',
      highlights: item.highlights,
      id: slugify(item.company),
      keywords: item.keywords,
      photos: item.photos,
      position: item.position,
      sectors: item.sectors,
      startDate: item.startDate,
    })),
  }
  return data
}

// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
export const cv = setIds(cleanObject(rawCv) as JsonResume)
