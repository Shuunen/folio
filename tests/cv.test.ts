import { expect, it } from 'vitest'
import { cleanObject, setIds } from '../data/cv'

it('cleanObject A', () => {
  expect(cleanObject({})).toMatchInlineSnapshot('{}')
})

it('cleanObject B', () => {
  expect(cleanObject({ maKey: 'Hello <br> there !' })).toMatchInlineSnapshot(`
    {
      "maKey": "Hello there !",
    }
  `)
})

it('cleanObject C', () => {
  expect(cleanObject({ wow: ['Damn !', '', 'I <b>love</b> this\n'] })).toMatchSnapshot()
})

it('cleanObject D', () => {
  expect(cleanObject({ level0: { wow: ['Damn !', 12, { hello: 'Que tal <image src="plop.jpg" /> ?' }, 'I <b>love</b> this\n'], another: { key: '' } } })).toMatchSnapshot()
})

it('setIds A', () => {
  expect(setIds({
    basics: { name: 'John Doe', label: 'Programmer', email: '', summary: 'A summary of John Doe...' },
    education: [],
    work: [{
      company: 'ACME Inc.',
      highlights: ['Started the company'],
      position: 'President',
      startDate: '2013-01-01',
      photos: [],
      keywords: [],
      endDate: '',
    }],
    interests: [],
    skills: [],
  })).toMatchSnapshot()
})
