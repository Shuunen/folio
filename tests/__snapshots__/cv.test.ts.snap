// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cleanObject C 1`] = `
{
  "wow": [
    "Damn !",
    "",
    "I love this",
  ],
}
`;

exports[`cleanObject D 1`] = `
{
  "level0": {
    "another": {
      "key": "",
    },
    "wow": [
      "Damn !",
      12,
      {
        "hello": "Que tal ?",
      },
      "I love this",
    ],
  },
}
`;

exports[`setIds A 1`] = `
{
  "education": [],
  "work": [
    {
      "company": "ACME Inc.",
      "endDate": "",
      "highlights": [
        "Started the company",
      ],
      "id": "acme-inc",
      "keywords": [],
      "photos": [],
      "position": "President",
      "sectors": [],
      "startDate": "2013-01-01",
    },
  ],
}
`;
